### PR TITLE
samples: wifi: shell: add IPv6-only variant

### PR DESCRIPTION
**Pull request by Damian-Nordic**
Add prj_ipv6only build variant with disabled IPv4, DHCP and static network configuration. Instead, it uses Router Advertisements for auto-configuration of IP addresses.

Signed-off-by: Damian Krolik <damian.krolik@nordicsemi.no>

### CODE REVIEW

File: samples/wifi/shell/prj_ipv6only.conf [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+# Copyright (c) 2022 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+# Configuration for IPv6-only build variant of the shell sample.
+# It relies on Router Advertisements rather than DHCP or static
+# configuration.
+
+# Wi-Fi driver
```
Comment (markaj-nordic) [inline samples/wifi/shell/prj_ipv6only.conf:11]: Can we consider adding `CONFIG_NRF_WIFI_LOW_POWER` here as well? We've had issues with this option enabled in Matter samples.

Comment (Damian-Nordic) [reply samples/wifi/shell/prj_ipv6only.conf:11]: We can, but it doesn't really work for me, so I wasn't sure we would like to provide an unstable config until it is investigated and fixed. If you think we would, I can add it.

Comment (krish2718) [reply samples/wifi/shell/prj_ipv6only.conf:11]: It is being tested, when it works it will be enabled in the main `prj.conf`

Comment (markaj-nordic) [reply samples/wifi/shell/prj_ipv6only.conf:11]: Ok, then let's just wait.

**Review by markaj-nordic — APPROVED** [review]

**Review by krish2718 — CHANGES_REQUESTED** [review]
You don't need to copy the original `prj.conf` file, just rename this to `overlay-ipv6only.conf` and then add new settings or overrides, this will be then merged with `prj.conf`

**Comment by Damian-Nordic** [comment]
> You don't need to copy the original `prj.conf` file, just rename this to `overlay-ipv6only.conf` and then add new settings or overrides, this will be then merged with `prj.conf`

The thing is that if you define two settings, say CONFIG_A and CONFIG_B, and `B depends on A` then even if you disable `A` in an overlay, you will get a build warning that `B` is modified although it depends on disabled symbol `A`. I wanted to avoid that. But if we accept the warning I can change it to the overlay.

**Comment by krish2718** [comment]
> > You don't need to copy the original `prj.conf` file, just rename this to `overlay-ipv6only.conf` and then add new settings or overrides, this will be then merged with `prj.conf`
> 
> The thing is that if you define two settings, say CONFIG_A and CONFIG_B, and `B` depends on `A` then even if you disable `A` in an overlay, you will get a build warning that `B` is modified although it depends on disabled symbol `A`. I wanted to avoid that. But if we accept the warning I can change it to the overlay.

Sure, but do we have such scenario here? If yes, then its fine.

**Comment by Damian-Nordic** [comment]
> Sure, but do we have such scenario here? If yes, then its fine.
Ok, we did, but I managed to unset string variables in the overlay, and the only left `NET_IF_UNICAST_IPV4_ADDR_COUNT` doesn't seem to be needed in prj.conf (it's set to the default).

**Comment by Damian-Nordic** [comment]
@krish2718 could you take another look?

**Review by krish2718 — APPROVED** [review]

**Pull request merged**